code/modules/mob/living/silicon/robot/robot.dm

```diff
@@ -1311,3 +1314,16 @@
 			locked = 1
 		notify_ai(1)
 		. = 1
+
+
+/mob/living/silicon/robot/proc/security_level(security_level = SEC_LEVEL_GREEN)
+	if(security_level == SEC_LEVEL_RED || security_level == SEC_LEVEL_DELTA)
+		if(designation == "Peacekeeper")
+			var/obj/item/borg/upgrade/reset/re = new
+			re.action(src)
+			pick_module("Security")
+	else
+		if(designation == "Security")
```

Comment:
if a server DOES allow secborgs as initial designation (outside of code red), this will turn them into peacekeepers too when code red ends